ingress-node-firewall/v1alpha1/ingressnodefirewall_types.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// IngressNodeFirewallICMPRule define ingress node firewall rule for ICMP and ICMPv6 protocols
+type IngressNodeFirewallICMPRule struct {
+	// ICMPType define ICMP Type Numbers (RFC 792).
```

Comment:
I would add some additional context here about examples of what possible values might be and what they would mean to an end user

Are you sure you want to allow all of the ICMP types? Is there a way to make this API nicer by allowing a subset and using human readable names rather than numbers?

Reply:
user can use any icmpv4 type and/or code based on https://www.rfc-editor.org/rfc/rfc792.html
and icmpv6 types and/or codes bases on https://www.rfc-editor.org/rfc/rfc4443.html

Reply:
Reading that list, it would seem there is a list of valid values of ICMPv4
```
Summary of Message Types

    0  Echo Reply

    3  Destination Unreachable

    4  Source Quench

    5  Redirect

    8  Echo

   11  Time Exceeded

   12  Parameter Problem

   13  Timestamp

   14  Timestamp Reply

   15  Information Request

   16  Information Reply
```
We can make an enum out of the numeric values, or could make it easier to understand, out of the textual values.

Reply:
we have types and codes we will endup with huge enum list one for v4 and another for v6 we already allow user entrying number for dstports and cidr so I wil ljust go with min and make sure its fits in 255 byte size